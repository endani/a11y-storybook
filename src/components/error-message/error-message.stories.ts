import type { Meta, StoryObj } from "@storybook/react"
import { ErrorMessage } from "./error-message"

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Example/ErrorMessage",
  component: ErrorMessage,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {},
  args: { children: "Ops! Something went wrong. Try again" },
} satisfies Meta<typeof ErrorMessage>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
  args: {},
}

export const WithColor: Story = {
  args: {
    color: "red",
  },
}
