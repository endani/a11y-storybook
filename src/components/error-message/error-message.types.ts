export interface ErrorMessageProps {
  children: string | React.ReactNode
  color?: string
}
