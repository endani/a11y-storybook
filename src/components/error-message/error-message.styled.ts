import styled from "styled-components"
import { ErrorMessageProps } from "./error-message.types"

export const ErrorMessageWrapper = styled.p<ErrorMessageProps>`
  font-family: "Nunito Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 400;

  color: ${({ color }) => (color ? color : "#000")};
`
