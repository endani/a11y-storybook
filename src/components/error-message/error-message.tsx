import { ErrorMessageWrapper } from "./error-message.styled"
import { ErrorMessageProps } from "./error-message.types"

const ErrorMessage = ({ children, color }: ErrorMessageProps) => (
  <ErrorMessageWrapper data-qa="error-message" color={color}>
    {children}
  </ErrorMessageWrapper>
)

export { ErrorMessage }
