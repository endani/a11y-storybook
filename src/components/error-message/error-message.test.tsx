import { render, screen } from "@testing-library/react"
import { ErrorMessage } from "./error-message"
import { ErrorMessageProps } from "./error-message.types"

const renderErrorMessage = ({ children, color }: ErrorMessageProps) =>
  render(<ErrorMessage color={color}>{children}</ErrorMessage>)

describe("ErrorMessage", () => {
  it("should render a ErrorMessage correctly", () => {
    const { container } = renderErrorMessage({
      children: "Awesome ErrorMessage",
    })

    expect(container).toMatchSnapshot()
    expect(screen.getByTestId("error-message")).toBeInTheDocument()
  })

  it("should render a ErrorMessage with custom color", () => {
    renderErrorMessage({
      children: "Awesome ErrorMessage",
      color: "#000",
    })

    expect(screen.getByTestId("error-message")).toHaveStyle("color: #000")
  })
})
