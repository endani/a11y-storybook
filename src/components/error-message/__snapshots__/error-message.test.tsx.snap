// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render a ErrorMessage correctly 1`] = `
<div>
  <p
    class="sc-aYaIB hymwfN"
    data-qa="error-message"
  >
    Awesome ErrorMessage
  </p>
</div>
`;
